lucetc/src/lib.rs

```diff
@@ -398,3 +410,18 @@ where
     }
     Ok(())
 }
+
+fn ldflags_default(target: &Triple) -> String {
+    use target_lexicon::OperatingSystem;
+
+    match target.operating_system {
+        OperatingSystem::Linux => "-shared",
+        OperatingSystem::MacOSX {
+            major: _,
+            minor: _,
+            patch: _,
+        } => "-dylib -dead_strip -export_dynamic -undefined dynamic_lookup",
+        _ => panic!("need to define default flags for OS"),
```

Comment:
Would you mind printing a more detailed error message? Think about the fact that most users have absolutely no experience with the code, and don't care about it. But they will see a `--target` option and will intuitively try to pass a triple. And then, "it doesn't work". The message needs to explain why and how to solve it.